src/main/java/seedu/address/model/task/UniqueTaskList.java

```diff
@@ -105,6 +107,72 @@ public void setTasks(List<? extends ReadOnlyTask> tasks) throws DuplicateTaskExc
         return FXCollections.unmodifiableObservableList(mappedList);
     }
 
+    /**
+     * Sorts person list by all persons by any field in ascending or descending order
+     * @param field
+     * @param order
+     */
+    //@@author charlesgoh
+    public void sortBy(String field, String order) {
+        //sortyBy first chooses the right comparator
+        Comparator<Task> comparator = null;
+
+        /**
+         * Comparators for the various fields available for sorting
+         */
+        Comparator<Task> priorityComparator = new Comparator<Task>() {
+            @Override
+            public int compare(Task o1, Task o2) {
+                return o1.getPriority().value.compareTo(o2.getPriority().value);
+            }
+        };
+
+        Comparator<Task> deadlineComparator = new Comparator<Task>() {
+            @Override
+            public int compare(Task o1, Task o2) {
+                return o1.getDeadline().value.compareTo(o2.getDeadline().value);
```

Comment:
This line can be problematic, because you are comparing the two deadlines by their String value, which does not guarantee sorting in chronological order. Changing "value" to "date" should resolve this issue.

Reply:
That's true! Thanks for pointing it out!